docs/_articles/en/user-guide/perf-enhancements.md

```diff
@@ -0,0 +1,17 @@
+---
+title: Performance enhancements
+lang: en
+---
+
+The performance enhancement changes allow you to run single file metadata deploys through a more efficient code path. These changes currently only support Apex Class metadata types. Additional metadata types will be supported in future releases.
+
+> NOTICE: Performance enhancement changes for single Apex Class file deploys is currently in beta. If you find any bugs or have feedback, [open a GitHub issue](./en/bugs-and-feedback).
+
+## Setup
+
+To enable this beta feature:
+
+1. Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS).
+1. Under Salesforce Feature Previews, select Experimental: Deploy Retrieve.
+
+In this beta release, performance enhancements will only occur when running the **SFDX: Deploy this Source to Org** command from an Apex Class.
```

Comment:
Should we specify which code paths are affected? ie. the right click menus and the command palette invokations?

Reply:
In both code paths the command name is `SFDX: Deploy this Source to Org`, I think we can add it or leave it as is. Whatever you think makes it clearer.